packages/components/src/Tree/Tree.tsx

```diff
@@ -194,10 +127,81 @@ const TreeLayout: FC<TreeProps> = ({
   )
 
   return (
-    <TreeContext.Provider value={{ border: isBorderEnabled, depth: nextDepth }}>
+    <TreeContext.Provider
+      value={{
+        border: hasBorder,
+        depth: depth + 1,
+        detailAccessory: hasDetailAccessory,
+        detailHoverDisclosure: hasDetailHoverDisclosure,
+      }}
+    >
       <TreeStyle depth={depth}>{internalAccordion}</TreeStyle>
     </TreeContext.Provider>
   )
 }
 
 export const Tree = styled(TreeLayout)``
+
+const generateTreeBorder = (depth: number, theme: Theme) => {
+  const {
+    colors,
+    space: { xxsmall, small },
+  } = theme
+
+  const itemBorderSize = '1px'
+  const itemPaddingSize = xxsmall
+  const indicatorIconSize = small
+  const indicatorGapSize = xxsmall
+  const depthSize = `${itemBorderSize} + ${itemPaddingSize} + (${indicatorIconSize} + ${indicatorGapSize}) * ${depth}`
+  const borderSpacer = `(${small} / 2) + ${depthSize}`
+
+  return css`
+    background: linear-gradient(
+      90deg,
+      transparent calc(${borderSpacer} - 1px),
+      ${colors.ui2},
+      transparent calc(${borderSpacer})
+    );
+  `
+}
+
+interface TreeBorderProps {
+  border?: boolean
+  depth: number
+}
+
+const TreeBorder = styled.div<TreeBorderProps>`
+  ${({ border, depth, theme }) => border && generateTreeBorder(depth, theme)}
+`
+
+const TreeStyle = styled.div<{ depth: number }>`
+  ${AccordionDisclosure} {
+    height: 25px;
+    padding: ${({ theme }) => theme.space.xxsmall};
+    padding-left: ${({ depth, theme }) =>
+      `calc(${theme.space.xxsmall} + (${theme.space.xxsmall} + ${theme.space.small}) * ${depth})`};
+  }
+
+  ${AccordionDisclosure} ${TreeItemPrimary} {
+    background-color: transparent;
+    padding: ${({ theme }) => theme.space.none};
+  }
+
+  ${AccordionDisclosure} ${TreeItem}:focus-within {
+    border-color: transparent;
+  }
+
+  ${TreeGroupLabel} {
+    padding-left: ${({ depth, theme }) =>
+      `calc(${theme.space.xxsmall} + (${theme.space.xxsmall} + ${
+        theme.space.small
+      }) * ${depth + 1})`};
```

Comment:
```suggestion
    padding-left: ${({
      depth,
      theme: {
        space: { small, xxsmall },
      },
    }) => `calc(${xxsmall} + (${xxsmall} + ${small}) * ${depth})`};
```

While this makes the de-structure a bit more complex  this change makes it easier to read-through the logic in play here.

Reply:
Resolved in 6692ebcce